src/main/java/com/commercetools/sync/services/impl/BaseService.java

```diff
@@ -288,12 +305,20 @@
       return CompletableFuture.completedFuture(keyToIdCache.asMap());
     }
 
-    final Consumer<Set<ResourceKeyId>> resultConsumer =
-        results -> results.forEach(resource -> keyToIdCache.putAll(keyToIdMapper.apply(resource)));
+    final List<List<String>> chunkedKeys = ChunkUtils.chunk(keysNotCached, CHUNK_SIZE);
```

Comment:
Do we need to chunk for graphql? It's sent in body as POST, so I don't think it's a problem here.

Reply:
As discussed, Yes we will need to do it for graphql.